### ISSUE TITLE
[BUG] When `cv = "prefit"`, number of classes between calibration set and fitted model can be different

### ISSUE THREAD

**Issue by vtaquet** [scikit-learn-contrib/MAPIE#116]
**Describe the bug**
For `MapieClassifier`, when `cv` is set to "prefit", the calibration set does not necessarily contain all the classes of the original dataset. Therefore, the base classifier could have been trained on a training set containing a larger number of classes. This discrepancy triggers a bug for the `cumulated_score` in the `fit` method.

**To Reproduce**
Simply run a simulation with a toy dataset in which the calibration set does not contain all the classes of the original dataset, with `cv="prefit"` and `method="cumulated_score"`. 

**Expected behavior**
This discrepancy can be fixed by using `sklearn.preprocessing.label_binarize` instead of ` sklearn.preprocessing.LabelBinarizer` in order to explicitely list the number of classes when "binarizing" the labels.
